tools/hist/fio-histo-log-pctiles.py

```diff
@@ -307,99 +298,61 @@ def get_pctiles(buckets, wanted, time_ranges):
-# returns a tuple of command line parameters
-# parameters have default values unless otherwise shown
-
-def parse_cli_params():
-    
-    # default values for input parameters
-
-    fio_version = 3        # we are using fio 3.x now
-    bucket_groups = None   # defaulting comes later
-    bucket_bits = 6        # default in fio 3.x
-    pctiles_wanted = [ 0, 50, 90, 95, 99, 100 ]
-    time_quantum = 1
-    output_unit = 'usec'
-
-    # parse command line parameters and display them
-    
-    argindex = 1
-    argct = len(sys.argv)
-    if argct < 2:
-        usage('must supply at least one histogram log file')
-    while argindex < argct:
-        if argct < argindex + 2:
-            break
-        pname = sys.argv[argindex]
-        pval = sys.argv[argindex+1]
-        if not pname.startswith('--'):
-            break
-        argindex += 2
-        pname = pname[2:]
-    
-        if pname == 'bucket-groups':
-            bucket_groups = int(pval)
-        elif pname == 'bucket-bits':
-            bucket_bits = int(pval)
-        elif pname == 'time-quantum':
-            time_quantum = int(pval)
-        elif pname == 'percentiles':
-            pctiles_wanted = [ float(p) for p in pval.split(',') ]
-        elif pname == 'output-unit':
-            if pval == 'msec' or pval == 'usec':
-                output_unit = pval
-            else:
-                usage('output-unit must be usec (microseconds) or msec (milliseconds)')
-        elif pname == 'fio-version':
-            if pval != '2' and pval != '3':
-                usage('invalid fio version, must be 2 or 3')
-            fio_version = int(pval)
-        else:
-            usage('invalid parameter name --%s' % pname)
+# this is really the main program
 
-    if not bucket_groups:
+def compute_percentiles_from_logs():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--fio-version", dest="fio_version", 
+        default="3", choices=[2,3], type=int, 
+        help="fio version (default=3)")
+    parser.add_argument("--bucket-groups", dest="bucket_groups", default="29", type=int, 
+        help="fio histogram bucket groups (default=29)")
+    parser.add_argument("--bucket-bits", dest="bucket_bits", 
+        default="6", type=int, 
+        help="fio histogram buckets-per-group bits (default=6 means 64 buckets/group)")
+    parser.add_argument("--percentiles", dest="pctiles_wanted", 
+        default="0 50 95 99 100", type=float, nargs='+',
+        help="fio histogram buckets-per-group bits (default=6 means 64 buckets/group)")
+    parser.add_argument("--time-quantum", dest="time_quantum", 
+        default="1", type=int,
+        help="time quantum in seconds (default=1)")
+    parser.add_argument("--output-unit", dest="output_unit", 
+        default="usec", type=str,
+        help="Latency percentile output unit: msec|usec|nsec (default usec)")
+    parser.add_argument("file_list", nargs='+')
+    args = parser.parse_args()
+    print(args)
```

Comment:
Is this debugging output? Should it be deleted?

Reply:
dang it.  you were right.  Will try to get rid of that next time around.